tools/utils.cc

```diff
@@ -72,7 +72,7 @@ void configure_tool_mode(app_template::seastar_options& opts, const sstring& log
 
 int tool_app_template::run_async(int argc, char** argv, noncopyable_function<int(const operation&, const boost::program_options::variables_map&)> main_func) {
     const operation* found_op = nullptr;
-    if (std::strcmp(argv[1], "--help") != 0 && std::strcmp(argv[1], "-h") != 0) {
+    if (std::strncmp(argv[1], "--help", 6) != 0 && std::strcmp(argv[1], "-h") != 0) {
```

Comment:
This if is a special case of all options that are a variant of `--help`. When this is detected, we don't want to attempt to parse `argv[1]` as a command name, it will fail. Instead we just print the generic help content (done below).
I will add a comment.